x/mongo/driver/topology/rtt_monitor.go

```diff
@@ -190,6 +195,28 @@ func min(samples []time.Duration, minSamples int) time.Duration {
 	return min
 }
 
+// percentile returns the specified percentile value of the slice of duration samples. Zero values
+// are not considered samples and are ignored. If no samples or fewer than minSamples are found
+// in the slice, percentile returns 0.
+func percentile(perc float64, samples []time.Duration, minSamples int) time.Duration {
+	// Convert Durations to float64s.
+	floatSamples := make([]float64, 0)
```

Comment:
Set the capacity of `floatSamples` to the length of `samples` so it only requires a single allocation.

E.g.
```go
floatSamples := make([]float64, 0, len(samples))
```